### PR TITLE
fix(injection): Some fixes for Deb->OCI migration of the injector

### PR DESCRIPTION
**Pull request by BaptisteFoy**
## Motivation
- Fixes version reporting for OCI on previously deb-only tests
- Hardcodes the socket paths so that tests work even during the deb->oci migration
  - Especially for agent in container
- Uses staging OCI injector for `dev` environment on previously deb-only tests

## Changes
Creates the config folder manually

## Workflow

1. ⚠️ Create your PR as draft ⚠️
2. Work on you PR until the CI passes (if something not related to your task is failing, you can ignore it)
3. Mark it as ready for review
    * Test logic is modified? -> Get a review from RFC owner. We're working on refining the `codeowners` file quickly.
    * Framework is modified, or non obvious usage of it -> get a review from [R&P team](https://dd.enterprise.slack.com/archives/C025TJ4RZ8X)

:rocket: Once your PR is reviewed, you can merge it!

🛟 [#apm-shared-testing](https://dd.enterprise.slack.com/archives/C025TJ4RZ8X) 🛟

## Reviewer checklist

* [ ] [Relevant label](https://github.com/DataDog/system-tests/blob/main/docs/CI/labels.md) (`run-parametric-scenario`, `run-profiling-scenario`...) are presents
* [ ] If PR title starts with `[<language>]`, double-check that only `<language>` is impacted by the change
* [ ] No system-tests internal is modified. Otherwise, I have the approval from [R&P team](https://dd.enterprise.slack.com/archives/C025TJ4RZ8X)
* [ ] CI is green, or failing jobs are not related to this change (and you are 100% sure about this statement)
* [ ] A docker base image is modified?
    * [ ] the relevant `build-XXX-image` label is present
* [ ] A scenario is added (or removed)?
    * [ ] Get a review from [R&P team](https://dd.enterprise.slack.com/archives/C025TJ4RZ8X)

### CODE REVIEW

File: utils/build/virtual_machine/provisions/host-auto-inject-install-script/auto-inject_host_script.yml [hunk 1]
```diff
@@ -7,4 +7,7 @@
         local_path: utils/build/virtual_machine/provisions/auto-inject/tracer_debug/debug_config.yaml
   remote-command: |
     bash execute_install_script.sh host 
-    sudo cp debug_config.yaml /etc/datadog-agent/inject/debug_config.yaml
\ No newline at end of file
+    sudo mkdir -p /var/run/datadog-installer
```
Comment (raphaelgavache) [inline utils/build/virtual_machine/provisions/host-auto-inject-install-script/auto-inject_host_script.yml:10]: this won't be needed anymore on Friday

File: utils/build/virtual_machine/provisions/host-auto-inject-install-script/auto-inject_host_script.yml [hunk 2]
```diff
@@ -7,4 +7,7 @@
...
+    sudo mkdir -p /opt/datadog/apm/inject/run
```
Comment (raphaelgavache) [inline utils/build/virtual_machine/provisions/host-auto-inject-install-script/auto-inject_host_script.yml:11]: (adding ctx) this is required as long as we have the mix between deb/rpm tracers, injector OCI and agent 
On complete of the OCI migration we can remove it, including the config below

**Review by raphaelgavache — APPROVED** [review]
lgtm!

File: utils/build/virtual_machine/provisions/host-auto-inject-install-script/auto-inject_host_script.yml [hunk 3]
```diff
@@ -7,4 +7,7 @@
...
+    printf "DD_APM_RECEIVER_SOCKET=/opt/datadog/apm/inject/run/apm.socket\nDD_DOGSTATSD_SOCKET=/opt/datadog/apm/inject/run/dsd.socket\nDD_USE_DOGSTATSD=true\n" | sudo tee /var/run/datadog-installer/environment
+    sudo cp debug_config.yaml /etc/datadog-agent/inject/debug_config.yaml
```
Comment (robertomonteromiguel) [inline utils/build/virtual_machine/provisions/host-auto-inject-install-script/auto-inject_host_script.yml:13]: I think this cp is a mistake. We don't need this configuration for host scenarios (we are configuring the auto inject using the env variables).
We need this file for container scenarios.
I think it was my mistake, and it's here for long time ago. I didn't realize until it start failing

**Review by robertomonteromiguel — CHANGES_REQUESTED** [review]

File: utils/build/virtual_machine/provisions/auto-inject/auto-inject-environment.yml [hunk 4]
```diff
@@ -8,6 +8,7 @@
   injection_major_version: apm
   deb_repo_name: beta
   rpm_repo_name: datadog-staging
+  site: datad0g.com
```
[the remark this answers is not in the document]
Comment (BaptisteFoy) [reply utils/build/virtual_machine/provisions/auto-inject/auto-inject-environment.yml:11]: It's the repository URL. That variable is poorly named - I'll change that

File: utils/build/virtual_machine/provisions/host-auto-inject-install-script/auto-inject_host_script.yml (hunk 3 above)
Comment (BaptisteFoy) [reply utils/build/virtual_machine/provisions/host-auto-inject-install-script/auto-inject_host_script.yml:13]: Sounds good, I'll remove it

Comment (BaptisteFoy) [reply utils/build/virtual_machine/provisions/host-auto-inject-install-script/auto-inject_host_script.yml:13]: Actually this looks required, without it the injection fails with 

```
{"level":"error","ts":1718881200.3642743,"caller":"process/main.go:187","msg":"error when parsing","error":"error loading config: open /etc/datadog-agent/inject/debug_config.yaml: no such file or directory","stacktrace":"main.main\n\t/go/src/github.com/DataDog/auto_inject/preload_go/cmd/process/main.go:187\nruntime.main\n\t/usr/local/go/src/runtime/proc.go:271"}
```

**Review by robertomonteromiguel — APPROVED** [review]

**Pull request merged**